Documentation/guides/internals/JavaJNI_Interop.md

```diff
@@ -0,0 +1,750 @@
+<!--toc:start-->
+- [Introduction](#introduction)
+- [Java <-> Managed interoperability overview](#java-managed-interoperability-overview)
+  - [Java Callable Wrappers (JCW)](#java-callable-wrappers-jcw)
+- [Registration](#registration)
+  - [Dynamic registration](#dynamic-registration)
+    - [Dynamic Java Callable Wrappers registration code](#dynamic-java-callable-wrappers-registration-code)
+    - [Dynamic Registration call sequence](#dynamic-registration-call-sequence)
+  - [Marshal methods](#marshal-methods)
+    - [Marshal Methods Java Callable Wrappers registration code](#marshal-methods-java-callable-wrappers-registration-code)
+    - [Marshal methods C# source code](#marshal-methods-c-source-code)
+    - [JNI requirements](#jni-requirements)
+    - [LLVM IR code generation](#llvm-ir-code-generation)
+    - [Assembly rewriting](#assembly-rewriting)
+      - [Wrappers for methods with non-blittable types](#wrappers-for-methods-with-non-blittable-types)
+      - [UnmanagedCallersOnly attribute](#unmanagedcallersonly-attribute)
+    - [Marshal Methods Registration call sequence](#marshal-methods-registration-call-sequence)
+<!--toc:end-->
+
+# Introduction
+
+At the core of `Xamarin.Android` is its ability to interoperate with
```

Comment:
Much as it pains me to say this, we should begin avoiding "Xamarin", *especially* for anything targeting .NET 8.  We should probably say:

> At the core of .NET Android is…